azurerm/resource_arm_application_insights_analytics_item.go

```diff
@@ -0,0 +1,219 @@
+		ID:      &itemID,
+		Name:    &name,
+		Type:    itemType,
+		Scope:   itemScope,
+		Content: &content,
+	}
+	if functionAlias != "" {
+		properties.Properties = &insights.ApplicationInsightsComponentAnalyticsItemProperties{
+			FunctionAlias: &functionAlias,
+		}
+	}
+
+	var itemScopePath insights.ItemScopePath
+	if itemScope == insights.ItemScopeUser {
+		itemScopePath = insights.MyanalyticsItems
+	} else {
+		itemScopePath = insights.AnalyticsItems
+	}
+	result, err := client.Put(ctx, resourceGroupName, appInsightsName, itemScopePath, properties, &overwrite)
+	if err != nil {
+		return fmt.Errorf("Error Putting Application Insights Analytics Item %s (Resource Group %s, App Insights Name: %s): %s", name, resourceGroupName, appInsightsName, err)
+	}
+
+	d.SetId(*result.ID)
+
+	return resourceArmApplicationInsightsAnalyticsItemRead(d, meta)
+}
+
+func resourceArmApplicationInsightsAnalyticsItemRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).appInsights.AnalyticsItemsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resourceGroupName := d.Get("resource_group_name").(string)
+	appInsightsID := d.Get("application_insights_id").(string)
+	scopeName := d.Get("scope").(string)
+	itemID := d.Id()
+
+	id, err := azure.ParseAzureResourceID(appInsightsID)
+	if err != nil {
+		return fmt.Errorf("Error parsing resource ID: %s", err)
+	}
+
+	appInsightsName := id.Path["components"]
+	name := d.Get("name").(string)
+
+	var scopePath insights.ItemScopePath
+	if scopeName == "user" {
+		scopePath = insights.MyanalyticsItems
+	} else {
+		scopePath = insights.AnalyticsItems
+	}
+	result, err := client.Get(ctx, resourceGroupName, appInsightsName, scopePath, itemID, name)
+	if err != nil {
+		return fmt.Errorf("Error Getting Application Insights Analytics Item %s (Resource Group %s, App Insights Name: %s): %s", name, resourceGroupName, appInsightsName, err)
+	}
+
+	d.Set("version", result.Version)
+	d.Set("content", result.Content)
+	d.Set("scope", string(result.Scope))
+	d.Set("type", string(result.Type))
+	d.Set("time_created", result.TimeCreated)
+	d.Set("time_modified", result.TimeModified)
+
+	if result.Properties != nil {
+		d.Set("function_alias", result.Properties.FunctionAlias)
+	}
+
+	return nil
+}
+
+func resourceArmApplicationInsightsAnalyticsItemDelete(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*ArmClient).appInsights.AnalyticsItemsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resourceGroupName := d.Get("resource_group_name").(string)
+	appInsightsID := d.Get("application_insights_id").(string)
+	scopeName := d.Get("scope").(string)
+	itemID := d.Id()
+
+	id, err := azure.ParseAzureResourceID(appInsightsID)
+	if err != nil {
+		return fmt.Errorf("Error parsing resource ID: %s", err)
+	}
+
+	appInsightsName := id.Path["components"]
+	name := d.Get("name").(string)
+
+	var scopePath insights.ItemScopePath
+	if scopeName == "user" {
```

Comment:
I think this can become:

```suggestion
	if scopeName == insights.ItemScopeUser {
```

Reply:
I had to convert to string, but have done this to avoid the magic string :-)